azurelinuxagent/ga/exthandlers.py

```diff
@@ -521,31 +526,37 @@ def handle_ext_handler(self, ext_handler, etag):
 
             return True
 
+        except ExtensionConfigError as error:
+            # Catch and report Invalid ExtensionConfig errors here to fail fast rather than timing out after 90 min
+            self.__handle_and_report_ext_handler_errors(ext_handler_i, error,
+                                                        report_op=WALAEventOperation.InvalidExtensionConfig,
+                                                        message="[Retryable Error] {0}".format(ustr(error)))
         except ExtensionUpdateError as error:
             # Not reporting the error as it has already been reported from the old version
             self.handle_ext_handler_error(ext_handler_i, error, error.code, report_telemetry_event=False)
         except ExtensionDownloadError as error:
-            self.handle_ext_handler_download_error(ext_handler_i, error, error.code)
+            msg = "Failed to download artifacts: {0}".format(ustr(error))
+            self.__handle_and_report_ext_handler_errors(ext_handler_i, error, report_op=WALAEventOperation.Download,
+                                                        message=msg)
         except ExtensionError as error:
             self.handle_ext_handler_error(ext_handler_i, error, error.code)
         except Exception as error:
             self.handle_ext_handler_error(ext_handler_i, error)
 
         return False
 
-    def handle_ext_handler_error(self, ext_handler_i, e, code=-1, report_telemetry_event=True):  # pylint: disable=C0103
-        msg = ustr(e)
+    @staticmethod
+    def handle_ext_handler_error(ext_handler_i, error, code=-1, report_telemetry_event=True):
+        msg = ustr(error)
         ext_handler_i.set_handler_status(message=msg, code=code)
 
         if report_telemetry_event:
             ext_handler_i.report_event(message=msg, is_success=False, log_event=True)
```

Comment:
At first I didn't see the difference between this method and `__handle_and_report_ext_handler_errors`. After all, they both set the handler status and report the error. The only difference I see is in the operation being reported to telemetry? Is there a way to combine these two methods since they're so similar?